src/sentry/profiles/task.py

```diff
@@ -153,29 +171,40 @@ def _symbolicate_profile(profile: Profile, project: Project) -> bool:
                 )
                 return True
 
-            # WARNING(loewenheim): This function call may mutate `profile`'s frame list!
-            # See comments in the function for why this happens.
-            raw_modules, raw_stacktraces, frames_sent = _prepare_frames_from_profile(profile)
+            platforms = get_profile_platforms(profile)
+            images = dict()
+            for platform in platforms:
+                images[platform] = get_debug_images_for_platform(
+                    profile["debug_meta"]["images"], platform
+                )
 
-            set_measurement("profile.frames.sent", len(frames_sent))
+            for platform in platforms:
+                # WARNING(loewenheim): This function call may mutate `profile`'s frame list!
+                # See comments in the function for why this happens.
+                profile["debug_meta"]["images"] = images[platform]
+                raw_modules, raw_stacktraces, frames_sent = _prepare_frames_from_profile(
+                    profile, platform
+                )
 
-            modules, stacktraces, success = run_symbolicate(
-                project=project,
-                profile=profile,
-                modules=raw_modules,
-                stacktraces=raw_stacktraces,
-            )
+                set_measurement(f"profile.frames.sent{platform}", len(frames_sent))
```

Comment:
instead of creating two measurements, can you not tag this with the platform?

Reply:
Looks like there can be 2 platforms here to support react native on iOS, namely, javascript and cocoa. 
```suggestion
                set_measurement(f"profile.frames.sent.{platform}", len(frames_sent))
```